pkg/controller/keyvault/ingress_secret_provider_class.go

```diff
@@ -103,7 +97,19 @@ func (i *IngressSecretProviderClassReconciler) Reconcile(ctx context.Context, re
 		},
 	}
 	logger = logger.WithValues("spc", spc.Name)
-	ok, err := i.buildSPC(ing, spc)
+
+	// Checking if we manage the ingress. All false cases without an error are assumed that we don't manage it
+	ok, err := i.ingressManager.IsManaging(ing)
+
+	if err != nil {
+		ok = false
+		err = fmt.Errorf("determining if ingress is managed: %w", err)
```

Comment:
why don't we just return early here and send the event? that's more idiomatic than waiting